src/wp-includes/media.php

```diff
@@ -1983,6 +2008,15 @@ function wp_image_use_alternate_mime_types( $image, $context, $attachment_id ) {
 				break 2;
 			}
 
+			/** This filter is documented in wp-includes/media.php. */
+			$filtered_image = apply_filters( 'wp_content_pre_replace_additional_image_source', $image, $attachment_id, $size_data, $name, $target_mime, $context );
+
+			if ( $filtered_image !== $image ) {
+				$image = $filtered_image;
+				unset( $metadata['sizes'][ $name ] );
```

Comment:
same question as above, do we need to reset this with the filtered image data instead of removing?